### PR TITLE
GraphQL subscriptions PR #5657

### PR DESCRIPTION
**Pull request by Abhishek-kumar09**
https://github.com/meshery/meshery/pull/5657

The actual changes intended to be made by https://github.com/meshery/meshery/pull/5657 is here. The problem arised due to merging the changes and then reverting again because the PR was not approved at that time. 

Rebasing to the base branch in that PR caused file to go away due to revert-merge.

### CODE REVIEW

**Review by leecalcote — DISMISSED** [review]

**Comment by leecalcote** [comment]
A couple of of build checks to look into, but the changes here are necessary.

File: internal/graphql/resolver/control-plane.go [hunk 1]
```diff
@@ -37,20 +37,26 @@ func (r *Resolver) listenToControlPlaneState(ctx context.Context, provider model
...
 				}
-				r.controlPlaneChannel <- status
-			case <-ctx.Done():
-				r.Log.Info("ControlPlane subscription stopped")
-				return
-			}
+
+			}(ctxID)
 		}
+
```
Comment (leecalcote) [inline internal/graphql/resolver/control-plane.go:59]:
```suggestion
```

**Comment by leecalcote** [comment]
Let's get a verbal review on this ... perhaps in Friday's meeting.

**Comment by leecalcote** [comment]
> Let's get a verbal review on this ... perhaps in Friday's meeting.

We missed this Friday. Let's shoot for Monday's meeting. :)

**Comment by leecalcote** [comment]
We missed Monday's meeting. Perhaps, Tuesday?

**Comment by Abhishek-kumar09** [comment]
@leecalcote @Revolyssup Ready to Merge?

**Comment by leecalcote** [comment]
I'm confused. I don't know that my request for a global MeshSync status is being met anymore.

**Review by Revolyssup — APPROVED** [review]

**Comment by Revolyssup** [comment]
@leecalcote  I think we can merge this. Because in terms of changes, there are two changes/enhancements :- 
1. A solved bug- Currently we have a single go channel which gets triggered everytime something is added or deleted in any cluster. This PR creates seperate channels for separate contexts so that addition in one cluster doesn't trigger anything in other. So this is kind of code cleanup/enhancement.
2. An extra redundant go channel is being used, that extra channel is being removed as it serves little to no purpose and adds unnecessary complexity.


So in totality, this PR removes more things than it adds and hence from my side can be safely merged.

**Pull request merged**